src/System.Text.Json/tests/Serialization/Stream.WriteTests.cs

```diff
@@ -121,48 +121,49 @@ public static async Task WritePrimitivesAsync()
         }
     }
 
-    public class TestStream : MemoryStream
+    public sealed class TestStream : Stream
     {
+        private readonly MemoryStream _stream;
+
+        public TestStream(int capacity) { _stream = new MemoryStream(capacity); }
+
+        public TestStream(byte[] buffer) { _stream = new MemoryStream(buffer); }
+
         public int TestFlushCount { get; private set; }
 
         public int TestWriteCount { get; private set; }
         public int TestWriteBytesCount { get; private set; }
-
         public int TestReadCount { get; private set; }
         public int TestRequestedReadBytesCount { get; private set; }
 
-        public TestStream(int capacity) : base(capacity) { }
-
-        public TestStream(byte[] buffer) : base(buffer) { }
+        public byte[] ToArray() => _stream.ToArray();
 
-        public override Task FlushAsync(CancellationToken cancellationToken)
+        public override void Flush()
         {
             TestFlushCount++;
-            return base.FlushAsync(cancellationToken);
+            _stream.Flush();
         }
 
-#if BUILDING_INBOX_LIBRARY
```

Comment:
I believe this was the only use case of this in the test projects. Can we remove it from:
https://github.com/dotnet/corefx/blob/a6f76f4f620cbe74821c6445af3f13e048361658/src/System.Text.Json/tests/System.Text.Json.Tests.csproj#L5-L8